web/src/app/environments/list/environments-list.component.html

```diff
@@ -84,7 +84,9 @@
             <img matListAvatar [src]="environment.logo" *ngIf="environment.logo" [alt]="environment.title">
             <h3 matLine
                 [class.tc-red-A700]="(!environment.latestPing.isValid && (environment.pings.valid + environment.pings.invalid) !== 0) || ['failBuild', 'failDeploy'].includes(environment.latestRelease.state)"
-                [class.tc-orange-A700]="![0, 100].includes(environment.progress.current)">
+                [class.tc-orange-A700]="![0, 100].includes(environment.progress.current)"
+                [class.tc-green-A700]="environment.type === 'build' && environment.latestRelease.state === 'finishBuild' || environment.type === 'build-deploy' && environment.latestRelease.state === 'finishDeploy'"
```

Comment:
Looks good. 

The `&&` conditions will need to be wrapped in brackets and in the `finishDeploy` state will also work for the environment type `deploy`

```
(environment.type === 'build' && environment.latestRelease.state === 'finishBuild') || (['deploy', 'build-deploy'].includes(environment.type) && environment.latestRelease.state === 'finishDeploy')
```

Reply:
I think we will need to apply the same logic to other places too, probably the environment view page also?

Reply:
Done :)